src/components/AttachmentPicker/index.native.js

```diff
@@ -253,63 +150,90 @@ class AttachmentPicker extends Component {
         });
     }
 
-    /**
-     * A generic handling when we don't know the exact reason for an error
-     *
-     */
-    showGeneralAlert() {
-        Alert.alert(this.props.translate('attachmentPicker.attachmentError'), this.props.translate('attachmentPicker.errorWhileSelectingAttachment'));
-    }
+    const [isVisible, setIsVisible] = useState(false);
+    const [focusedIndex, setFocusedIndex] = useState(-1);
+    const [menuItemData, setMenuItemData] = useState([
+        {
+            icon: Expensicons.Camera,
+            textTranslationKey: 'attachmentPicker.takePhoto',
+            pickAttachment: () => showImagePicker(launchCamera),
+        },
+        {
+            icon: Expensicons.Gallery,
+            textTranslationKey: 'attachmentPicker.chooseFromGallery',
+            pickAttachment: () => showImagePicker(launchImageLibrary),
+        },
+    ]);
```

Comment:
I prefer to keep such hooks at the very top of a component, to keep them in one place.

Reply:
So I've moved **isVisible** and **focusedIndex** to the top but I am wondering about **menuItemData** as it uses **showImagePicker**. That means the function should be defined before it (as it is now) or I can move it to the top and surpass the eslint rule. Which one do we prefer?

Reply:
Yeah, in this case feel free to have function first. AFAIK, we are going to tweak this initial value.